modules/long-arithmetic/src/long_arithmetic.cpp

```diff
@@ -0,0 +1,285 @@
+            this->negFlag = true;
+        } else {
+            this->negFlag = false;
+        }
+
+        for (int64_t i = str.length(); i > 0; i -= 9) {
+            if (i < 9)
+                this->storage.push_back(atoi(str.substr(0, i).c_str()));
+            else
+                this->storage.push_back(atoi(str.substr(i - 9, 9).c_str()));
+        }
+
+        this->delZeros();
+    }
+}
+
+void bigInt::delZeros() {
+    while (this->storage.size() > 1 && this->storage.back() == 0) {
+        this->storage.pop_back();
+    }
+
+    if (this->storage.size() == 1 && this->storage[0] == 0) this->
+        negFlag = false;
+}
+
+const bool bigInt::operator==(const bigInt& right) const {
+    bigInt left = *this;
+    if (left.negFlag != right.negFlag) return false;
+    if (left.storage.empty()) {
+        if (right.storage.empty() || (right.storage.size() == 1 &&
+            right.storage[0] == 0)) return true;
+        else
+            return false;
+    }
+
+    if (right.storage.empty()) {
+        if (left.storage.size() == 1 && left.storage[0] == 0) return true;
+        else
+            return false;
+    }
+
+    if (left.storage.size() != right.storage.size()) return false;
+    for (size_t i = 0; i < left.storage.size(); ++i) if
+        (left.storage[i] != right.storage[i]) return false;
+
+    return true;
+}
+
+const bigInt bigInt::operator +() const {
+    return bigInt(*this);
+}
+
+const bigInt bigInt::operator -() const {
+    bigInt copy(*this);
+    copy.negFlag = !copy.negFlag;
+    return copy;
+}
+
+const bool bigInt::operator<(const bigInt& right) const {
+    bigInt left = *this;
+    if (left.negFlag) {
+        if (right.negFlag) return ((-right) < (-left));
+        else
+            return true;
+    } else {
+        if (right.negFlag) {
+            return false;
+        } else {
+            if (left.storage.size() != right.storage.size()) {
+                return left.storage.size() < right.storage.size();
+            } else {
+                for (int64_t i = left.storage.size() - 1; i >= 0; --i) {
+                    if (left.storage[i] != right.storage[i]) return
+                        left.storage[i] < right.storage[i];
+                }
+                return false;
+            }
+        }
+    }
+}
+
+const bool bigInt::operator!=(const bigInt& right) const {
+    return !(*this == right);
+}
+
+const bool bigInt::operator<=(const bigInt& right) const {
+    return (*this < right || *this == right);
+}
+
+const bool bigInt::operator>(const bigInt& right) const {
+    return !(*this <= right);
+}
+
+const bool bigInt::operator>=(const bigInt& right) const {
+    return !(*this < right);
+}
+
+bigInt bigInt::operator+(const bigInt& right) const {
+    bigInt left = *this;
+    if (left.negFlag) {
+        if (right.negFlag) return -(-left + (-right));
+        else
+            return right - (-left);
+    } else {
+        if (right.negFlag) return left - (-right);
+    }
```

Comment:
почему нельзя так? не только здесь, но и в других операторах
```suggestion
    if (left.negFlag && right.negFlag) {
        return left - right;
    } else if (right.negFlag) {
        return left + right;
    }
```

Reply:
Потому что в крайних случаях вылезает ошибка с необработанным исключением. Например, если рассмотреть такие выражения:
 `bigInt c("999999999999999999999"), b("99999999999999999999");
	cout << c + b << endl;
	cout << -c + (-b) << endl;
	cout << -c + b << endl;
cout << c + (-b) << endl;`
Аналогично с оператором "-"